torch/jit/_backends.py

```diff
@@ -0,0 +1,63 @@
+"""Backend API
+This module contains helper functions and utilities for the JIT backend
+API.
+"""
+
+from torch.jit._script import (
+    RecursiveScriptModule,
+)
+from typing import Callable, List
+
+
+# Type for functions that lower RecursiveScriptModules to JIT backends.
+ToBackendFnTy = Callable[
+    [RecursiveScriptModule], RecursiveScriptModule
+]
+
+
+def selective_to_jit_backend(
+    module: RecursiveScriptModule,
+    to_backend_fn: ToBackendFnTy,
+    modules_to_lower: List[str],
+):
+    """
+    Selectively lower modules in a module hierarchy to a JIT backend.
+
+
+    Arguments:
+        module: The RecursiveScriptModule at the root of the module hierachy that
+                    is to be selectively lowered.
+        to_backend_fn:  Function that will be called on each submodule in modules_to_lower
+                            to lower it to a JIT backend.
+        modules_to_lower: List of fully qualified names of submodules to selectively
+                            lower.
+    """
+    base = ""
+    selective_to_jit_backend_impl(module, to_backend_fn, modules_to_lower, base)
+
+
+def selective_to_jit_backend_impl(
+    module: RecursiveScriptModule,
+    to_backend_fn: ToBackendFnTy,
+    modules_to_lower: List[str],
+    base: str,
+) -> RecursiveScriptModule:
+    # For each submodule:
+    for name, submodule in module._modules.items():
```

Comment:
In general, we should implement logic like this is in C++, to avoid increasing the surface area of the Python bindings and re-implementing functionality like `remapTypes`.